### PR TITLE
using gufe Settings in RBFE Protocol

### PR DESCRIPTION
**Pull request by richardjgowers**

### CODE REVIEW

File: openfe/tests/protocols/test_openmm_equil_protocols.py
Comment (IAlibay) [inline openfe/tests/protocols/test_openmm_equil_protocols.py:1]: Would it be possible to have a test that checks the default settings against a serialised version of the settings?

My take here is that whilst we _should_ know that things are changing upstream, it would be easy for us to forget to change / review its impact downstream. At least if it fails here we can go "yeah ok, that's fine" (or not).

**Review by IAlibay — CHANGES_REQUESTED** [review]
Still digging upstream for a few things but here's a first comment - we need to be more vigilant in picking up cases where settings changes happen, otherwise we'll one day walk into a minefield.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Still digging upstream for a few things but here's a first comment - we need to be more vigilant in picking up cases where settings changes happen, otherwise we'll one day walk into a minefield.

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:529]: this is a good moment to update this:
```suggestion
                    forcefield={'protein': 'amber/ff14SB.xml',
                                'ligand': 'openff-2.0.0.offxml',
                                'solvent': 'amber/tip3p_standard.xml'
                                'ions': 'amber/tip3p_HFE_multivalent.xml',},
```

Comment (IAlibay) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:529]: Indeed those are the forcefield settings found under `OpenMMSystemGeneratorFFSettings`.

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:517]: Why are we setting forcefields here and then again below  under protocol settings but using twto different objects?

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:266]: It might make more sense to just move this to integrator settings (renamed as `barostat_frequency`), in reality this is just a "how often should my integration loop stop and do an MC barostat move).

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:765]: switch to `forcefield_settings.small_molecule_forcefield`?

**Review by IAlibay — CHANGES_REQUESTED** [review]
I'm not going to block over the restructuring / merging of leftover setting classes (that can be dealt with later if necessary).

I am however going to block over the duplication of settings between what gets defined under `settings.forcefield_settings`, and elsewhere. That's probably a nightmare waiting to happen.

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:1086]:
```suggestion
            'repeat_id': self._inputs['repeat_id'],
            'generation': self._inputs['generation'],
```

**Review by IAlibay — CHANGES_REQUESTED** [review]
found this out the hard way this evening

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:376]: Is there a way to do this at the validator level? It feels like there should be, but I couldn't find a way to use a validator to validate across two pydantic classes as attributes of a model.

Comment (IAlibay) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:376]: The answer _should_ be to use a root_validator in the pydantic settings child class.

_However_, applying validators on nested models is all kinds of broken in pydantic right now (it's meant to be fixed in the v2.0 release). We might just have to keep there here until v2.0 gets released properly.

Comment (richardjgowers) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:376]: yeah I think we won't be able to get validators to do all the heavy lifting, but they'll at least do a good level of sanity checking

**Comment by IAlibay** [comment]
Requires #293 for CI to pass

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (IAlibay) [inline openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:354]: For tomorrow - I just realised, removing these is probably going to make it impossible to look at the docstring programmattically right? Should we keep them in and then also have the inline docstrings?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Going to block on this question.

**Comment by IAlibay** [comment]
This is how it looks if you interrogate a settings class right now:

image

File: openfe/protocols/openmm_rbfe/__init__.py
Comment (richardjgowers) [inline openfe/protocols/openmm_rbfe/__init__.py:12]: since Protocol depends on Settings, would it make more sense to import this first?

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (richardjgowers) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:381]: or do we call it `-ffast-science`?

Comment (IAlibay) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:381]: I think this definitely falls on the "this is a terrible idea" line. However, we can probably do a major review of how we validate things for 0.8?

File: openfe/protocols/openmm_rbfe/__init__.py
Comment (IAlibay) [reply openfe/protocols/openmm_rbfe/__init__.py:12]: do you mean moving it up the import list? sure

Comment (richardjgowers) [reply openfe/protocols/openmm_rbfe/__init__.py:12]: yes, it looks like a circular import?  I guess they don't exist any more

File: openfe/protocols/openmm_rbfe/equil_rbfe_methods.py
Comment (richardjgowers) [reply openfe/protocols/openmm_rbfe/equil_rbfe_methods.py:381]: yeah, probably a fun chat to have, how do we want the safety rails to behave

**Review by IAlibay — APPROVED** [review]
happy with the status of this, will defer to Richard for approval / merge

**Comment by IAlibay** [comment]
```
Object of type RelativeLigandProtocolSettings is not JSON serializable
```

:/ yeah I'm giving up on trying to get my mush brain around that one. Settings seem to serialise fine, but maybe I'm missing something obvious.

**Comment by richardjgowers** [comment]
right I think we're gucci, I'll merge this once green

**Pull request merged**
